pkg/scheduler/backend/api_dispatcher/call_queue_test.go

```diff
@@ -26,21 +26,44 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/component-base/metrics/testutil"
 	fwk "k8s.io/kube-scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/metrics"
 )
 
+func init() {
+	metrics.Register()
+}
+
+func resetMetrics() {
+	metrics.AsyncAPICallsTotal.Reset()
+	metrics.AsyncAPICallDuration.Reset()
+	metrics.AsyncAPIPendingCalls.Reset()
+}
+
 var queuedAPICallCmpOpts = []cmp.Option{
 	cmp.AllowUnexported(queuedAPICall{}, mockAPICall{}),
 	// Comparison of function fields is not reliable, so they are ignored.
 	cmpopts.IgnoreFields(mockAPICall{}, "executeFn", "mergeFn", "syncFn"),
 }
 
-// verifyQueueLen is a test helper to check the length of the callsQueue.
-func verifyQueueLen(t *testing.T, cq *callQueue, len int) {
+// verifyQueueState is a test helper to check both the queue length and pending call metrics.
+func verifyQueueState(t *testing.T, cq *callQueue, expectedQueueLen int, expectedPendingCalls map[string]int) {
 	t.Helper()
 
-	if got := cq.callsQueue.Len(); got != len {
-		t.Errorf("Expected callsQueue to have %d item(s), but has %d", len, got)
+	// Check queue length
+	if got := cq.callsQueue.Len(); got != expectedQueueLen {
+		t.Errorf("Expected callsQueue to have %d item(s), but has %d", expectedQueueLen, got)
+	}
+
+	// Check pending call metrics
+	for callType, expected := range expectedPendingCalls {
+		value, err := testutil.GetGaugeMetricValue(metrics.AsyncAPIPendingCalls.WithLabelValues(callType))
+		if err != nil {
+			t.Errorf("Failed to get pending calls metric for %s: %v", callType, err)
+		} else if value != float64(expected) {
```

Comment:
Let's do:
```suggestion
		} else if int(value) != expected {
```
And log the int